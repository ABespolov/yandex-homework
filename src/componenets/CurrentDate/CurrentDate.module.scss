@import "../../mixins";

.date{
  text-align: right;
  @include smoothResize(font-size, 16, 18);
}
