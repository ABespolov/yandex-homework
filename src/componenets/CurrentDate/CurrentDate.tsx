import React from "react";
import {distanceInWords, compareAsc} from 'date-fns';
import ru from 'date-fns/locale/ru';
import styles from "./CurrentDate.module.css";

interface CurrentDate {
  created: number;
}

export const CurrentDate: React.FC<CurrentDate> = ({created = 0}) => {
  const date = new Date(1970, 0, 1);
  date.setMilliseconds(created);
    const result = distanceInWords(
      new Date(),
      date,
      {locale: ru}
    );
  return (
    <div className={styles.date}>
      {`${result} назад`}
    </div>
  );
};
