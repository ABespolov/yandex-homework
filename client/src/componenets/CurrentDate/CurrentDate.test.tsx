import {configure, shallow} from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import {CurrentDate} from "./CurrentDate";

configure({ adapter: new Adapter() });

describe("Current date:", () => {
    it("Current date render: ", () => {
        const component = shallow(<CurrentDate created={12345678} />);
        expect(component).toMatchSnapshot();
    });
});
