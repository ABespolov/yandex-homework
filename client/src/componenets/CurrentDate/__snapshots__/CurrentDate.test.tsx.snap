// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current date: Current date render:  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentDate
    created={12345678}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "больше 49 лет назад",
      "className": "date",
    },
    "ref": null,
    "rendered": "больше 49 лет назад",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "больше 49 лет назад",
        "className": "date",
      },
      "ref": null,
      "rendered": "больше 49 лет назад",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
